### ISSUE TITLE
[DataFrame] DataFrame operations mutate original DataFrame

### ISSUE THREAD

**Issue by luisquintanilla** [dotnet/corefxlab#2734]
**DataFrame API Version: 1.0.0-e190910-1**
**.NET Core Version: 3.0**

Given the following code:

``` csharp
// Read data
DataFrame df = DataFrame.ReadCsv("taxi-fare-train.csv");

// Show column names
Console.WriteLine($"Column Names: {String.Join(",", df.Columns)}");
``` 

The output below is produced

```text
Column Names: vendor_id,rate_code,passenger_count,trip_time_in_secs,trip_distance,payment_type,fare_amount
```

After using the `AddSuffix` method

```csharp
// Add suffix to df
df.AddSuffix("_test");
Console.WriteLine(String.Join(",", df.Columns));
```

The result is the following

```text
Suffixed Column Names: vendor_id_test,rate_code_test,passenger_count_test,trip_time_in_secs_test,trip_distance_test,payment_type_test,fare_amount_test
```

Then, when trying to access a column by its original name

```csharp
Console.WriteLine(df["fare_amount"]);
```

It throws an `ArgumentException`.

```text
System.ArgumentException: 'Column does not exist (Parameter 'columnName')'
```

This makes sense because as shown after using the `AddSuffix` method, the names of the original DataFrame have changed. Therefore the correct way to access the column now is 

```csharp
Console.WriteLine(df["fare_amount_test"]);
```

That being said, the result of the evaluation of `AddSuffix` changes the values in the original DataFrame. Is that the intent? I would expect that performing operations on a DataFrame, although they can still be evaluated won't have an impact on the original DataFrame. If I want to persist the changes / transformations I'd have to assign it to a new DataFrame for those changes to be materialized

i.e.

```csharp
DataFrame suffixedDf = df.AddSuffix("_test");
```

**Comment by pgovind** [comment]
You're quite right @luisquintanilla. When I initially wrote this code, I assumed nobody would want to clone a big DataFrame just to change column names. However, given how the rest of the API looks, `AddSuffix` method should've returned a new DataFrame. I filed a bug [here](https://github.com/dotnet/corefxlab/issues/2739).

**Comment by pgovind** [comment]
Fixed with #2740
